packages/@datapull/jsonpath-multistep-transformer/index.js

```diff
@@ -0,0 +1,80 @@
+var jp = require('jsonpath');
+
+/**
+ * Example:
+ *
+ * source tree:
+
+ {
+   Results: [
+    {
+       value1: "A"
+       value2: "B",
+       subResults: [
+         { value3: "C" },
+         { value3: "D" },
+       ]
+    }
+   ]
+ }
+
+ * configure the transform step:
+ *
+     transform:
+       - jsonpath-multistep:
+       steps:
+       - query: $.Results.*
+       - query: $.subResults.*
+         captureFromPreviousStep:
+           value1: $.value1
+
+ * The result of transformation will be:
+
+   [ { value1: "A", value3: "C" }, { value1: "A", value3: "D" } ]
+
+ */
+class JsonPathMultistepTransformer {
+  constructor(config) {
+    console.log('[JsonPathMultistepTransformer]', config);
+    this.config = config;
+  }
+  transform(data) {
+    let selectedData = data;
+
+    this.config.steps.forEach(step => {
+      // if working with list of items, treat each item as a root of a separate tree
+      if (Array.isArray(selectedData)) {
+        let r = [];
+        selectedData.forEach(parent => {
+          let children = jp.query(parent, step.query);
+
+          // capture requested values from parent object:
+          if (step.captureFromPreviousStep) {
```

Comment:
I'm seriously confused what's going in the following loop. Could you elaborate with a comment (in the code)?